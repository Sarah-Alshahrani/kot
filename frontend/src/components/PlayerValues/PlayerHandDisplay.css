.player_hand_details {
    justify-content: center;
    display: flex;
    flex-direction: column;
    border-width: thin
}

.img-hand {
    margin: 3px;
    height: 55px
}