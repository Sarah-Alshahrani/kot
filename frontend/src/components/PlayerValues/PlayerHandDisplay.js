import React from "react";
import Hand from "./../Images/icons8-red-card-80.png";
import "./PlayerHandDisplay.css";
import PlayerCard from "./PlayerCard";

class PlayerHandDisplay extends React.Component {
  constructor(props, context) {
    super(props, context);

    let cards = this.updateCards(this.props.data);
    this.state = {
      username: this.props.currentUser,
      gameRoom: this.props.currentRoom,
      data: cards
    };
  }

  updateCards(data) {
    let cards = {};
    if (this.props.data) {
      try {
        cards = JSON.parse(data);
      } catch (e) {}
    }
    return cards;
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.data !== prevProps.data) {
      let cards = this.updateCards(this.props.data);
      this.setState({
        data: cards
      });
    }
  }

  render() {
    if (this.state.data && this.state.data.length > 0) {
      return (
        <div>
          <img className="img-hand" alt="" src={Hand}></img>
          <container className="player_hand_details">
            {this.state.data.map((entry, index) => (
              <PlayerCard
                key={index}
                name={entry.name}
                cost={entry.cost}
                effect={entry.effect}
                footnote={entry.footnote}
              />
            ))}
          </container>
        </div>
      );
    } else {
      return (
        <div>
          <img className="img-hand" alt="" src={Hand}></img>
          No Cards...
        </div>
      );
    }
  }
}
export default PlayerHandDisplay;
