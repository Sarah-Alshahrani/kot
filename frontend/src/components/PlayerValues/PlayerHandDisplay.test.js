import React from "react";
import { shallow } from "enzyme";

import PlayerHandDisplay from "./PlayerHandDisplay";

describe("Verify PlayerHandDisplay layout template", () => {
  const test_player_details = [
    {
      name: "howdy",
      cost: 1,
      effect: 0,
      footnote: 2
    },
    {
      name: "ya",
      victory_points: 5,
      health: 5,
      footnote: 4
    },
    {
      name: "all",
      cost: 10,
      health: 10,
      footnote: 6
    }
  ];

  const json_test_player_details = JSON.stringify(test_player_details);

  const test_username = "test_user";
  const test_roomname = "test_room";

  const expected_state_values = {
    username: test_username,
    gameRoom: test_roomname,
    data: test_player_details
  };

  it("Verify parameter provided PlayeValues are rendered correctly", () => {
    const component = shallow(
      <PlayerHandDisplay
        currentUser={test_username}
        currentRoom={test_roomname}
        data={json_test_player_details}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("Verify state transfers via props as expected", () => {
    const wrapper = shallow(
      <PlayerHandDisplay
        currentUser={test_username}
        currentRoom={test_roomname}
        data={json_test_player_details}
      />
    );
    expect(wrapper.state()).toEqual({ ...expected_state_values }); // passed
  });

  it("verify default states are as expected", () => {
    const wrapper = shallow(
      <PlayerHandDisplay
        currentUser={test_username}
        currentRoom={test_roomname}
        data={JSON.stringify([])}
      />
    );

    const expected_state_values_with_no_player_details = {
      username: test_username,
      gameRoom: test_roomname,
      data: []
    };

    expect(wrapper.state()).toEqual({
      ...expected_state_values_with_no_player_details
    });
  });
});
