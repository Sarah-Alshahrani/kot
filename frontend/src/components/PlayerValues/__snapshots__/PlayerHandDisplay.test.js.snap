// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify PlayerHandDisplay layout template Verify parameter provided PlayeValues are rendered correctly 1`] = `
<div>
  <img
    alt=""
    className="img-hand"
    src="icons8-red-card-80.png"
  />
  <container
    className="player_hand_details"
  >
    <PlayerCard
      cost={1}
      effect={0}
      footnote={2}
      key="0"
      name="howdy"
    />
    <PlayerCard
      footnote={4}
      key="1"
      name="ya"
    />
    <PlayerCard
      cost={10}
      footnote={6}
      key="2"
      name="all"
    />
  </container>
</div>
`;
